gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/computer/VertexProgram.java

```diff
@@ -219,13 +234,28 @@ public default void workerIterationEnd(final Memory memory) {
      * @param <V>           The vertex program type
      * @return the newly constructed vertex program
      */
+    @Deprecated
```

Comment:
you need a "@deprecated" javadoc here.

Reply:
Done in new commit. Will push with other changes.